lib/snowflex/ecto/ecto_adapter.ex

```diff
@@ -31,14 +31,19 @@ defmodule Snowflex.EctoAdapter do
 
   defp binary_encode(raw), do: {:ok, Base.encode16(raw)}
 
+  def decimal_decode(nil), do: {:ok, nil}
   def decimal_decode(dec) when is_binary(dec), do: {:ok, Decimal.new(dec)}
   def decimal_decode(dec) when is_float(dec), do: {:ok, Decimal.from_float(dec)}
 
+  defp int_decode(nil), do: {:ok, nil}
   defp int_decode(int) when is_binary(int), do: {:ok, String.to_integer(int)}
   defp int_decode(int), do: {:ok, int}
 
+  defp time_decode(decode), do: {:ok, nil}
```

Comment:
was it meant nil instead of decode here?

Reply:
Yeah, this perplexed me a bit too, even if its not totally necessary, I think if I'm understanding correctly the explicit match would be good here for clarity.

Reply:
Ops! fixed 😵‍💫